Bottom: 26f20723c47beedc39ab98795ee82eeb4cdc3a08
Top:    9d0a70995d880269781042daa6a62d92b3894a12
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-08 01:03:34 +0100

Refresh of re-write-ghid-set-layer

---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 27bd030..4ecd420 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -67,27 +67,26 @@ hid_gc_struct;
 
 static void draw_lead_user (render_priv *priv);
 
+/* Compute group visibility based upon on copper layers only */
+static bool
+is_layer_group_visible (int group)
+{
+  int entry;
+  for (entry = 0; entry < PCB->LayerGroups.Number[group]; entry++)
+    {
+      int layer_idx = PCB->LayerGroups.Entries[group][entry];
+      if (layer_idx >= 0 && layer_idx < max_copper_layer &&
+          LAYER_PTR (layer_idx)->On)
+        return true;
+    }
+  return false;
+}
 
 int
 ghid_set_layer (const char *name, int group, int empty)
 {
   if (group >= 0 && group < max_group)
-    {
-      int layer_idx;
-      int entry;
-      for (entry = 0; entry < PCB->LayerGroups.Number[group]; entry ++)
-	{
-	  layer_idx = PCB->LayerGroups.Entries[group][entry];
-	  /* Only return TRUE for visible non-silk layers.
-	   * NB: Silk layers ocupy layer_idx
-	   * max_copper_layer and max_copper_layer + 1
-	   */
-	  if (layer_idx >= 0 && layer_idx < max_copper_layer &&
-	      PCB->Data->Layer[layer_idx].On)
-	    return TRUE;
-	}
-      return FALSE;
-    }
+    return is_layer_group_visible (group);
 
   /* If we didn't hit a match above, group is being used as a
      special symbolic layer type, and will be negative. */
